README.md

```diff
@@ -1,74 +1,177 @@
+# Query Types
 
-### Fields:
+TellorX is designed to support arbitrary query types (`QueryType`).  A `QueryType` can have an arbitrary response type, specified by a structured
+ABI type string.  
 
-#### Question ID
+The following `QueryType`s are currently supported:
 
-    (name of file) can be guessed, but ultimately will be filled in by team.
+- LegacyRequest
+- TextQuery
+- CoinPrice (Under development)
 
-#### Query ID: 
+A `QueryType` can have multiple Query `parameters` that specify details of the query request (e.g. the token symbol for a `CoinPrice` query).
 
-    bytes32 query ID or structure for creating the ID (e.g. what to hash)
+Please submit a PR to this repository or contact the team if you would like to define a new `QueryType`. 
 
+# Query Descriptors
 
-#### Data Specifications
+Queries are uniquely identified by a special JSON string called the Query `Descriptor`.  In order to properly interact with the oracle, the `Descriptor` string must be provided *exactly* as defined by the [Query Descriptor Specification](#query-descriptor-specifications).
 
-    Technical Specifications of the value(s) requested
+The following is an example descriptor for a legacy query.  The `type` of query is a `LegacyRequest`.  The `LegacyRequest` query has a single integer parameter, called `legacy_id`.
 
+    {"type":"LegacyRequest","legacy_id":1}
 
-#### Description
+For further details, refer to [Query Descriptor Specifications](#query-descriptor-specifications) and the `QueryType` specifications for
+the query of interest.
 
-    Lamen terms description of the ID and it's implications
 
+# Requirements for QueryType Specifications
 
-#### Granularity
+Every `QueryType` Specification shall provide the following:
 
-    Multiplier to use (if numeric) for putting on-chain
+## Name
 
+The unique name of the QueryType (e.g. `LegacyRequest`)
+Further details of the `QueryType` specification shall be provided in a file named: 
 
-#### Example Value (w/ date)
+    <query-type-name>.md
 
-    An example piece of data (e.g. "9/27/21 - 9000000")
+## Description
 
-#### Dispute Considerations
+A general description of the QueryType, including its purpose and suggested use cases.
 
-    Considerations that parties should consider before reporting the data w/ regard to what will be considered a valid dispute
+## Query Parameters
 
-#### Notes
+A description of the query parameters shall include:
 
-    Any further notes or thoughts on the feed
+- A table defining the following for each `parameter`:
+  - `Name`
+  - `Description`
+  - `Data Type`
+  - `Value Specifications`
+- The parameter order required by the query `Descriptor`.
+- Valid Parameter Sets
+  - A list of valid parameter sets that are expected to be supported by reporters
+- Active Parameter Sets (optional)
+  - A list of parameter sets that are *actively* supported by current reporters
 
 
-## Maintainers <a name="maintainers"> </a> 
+## Response Type
```

Comment:
The response type is part of the query specification, and is not part of the descriptor.

Reply:
not sure what you mean by that

Reply:
It sounded like you were asking if the response type is part of the json descriptor.  It is not.  Rather, it will be required to be in the query specification document.

Reply:
yes this is what I meant and thanks for the clarification